components/molecule/pagination/src/index.js

```diff
@@ -75,6 +78,12 @@ const MoleculePagination = ({
   const isHidePrev = hideDisabled && !prevPage
   const isHideNext = hideDisabled && !nextPage
 
+  const linkProps = pageNumber =>
+    typeof linkFactory === 'function' && {
+      link: true,
+      linkFactory: linkFactory(pageNumber)
```

Comment:
It's not a good idea to give to `linkFactory` the responsibility to manage de URL, I suggest to better move the following functionality to this component: https://github.com/SUI-Components/adevinta-spain-components/blob/master/components/pagination/basic/src/sui-pagination-item.js#L36. So, you'll need to define 3 new props: `linkFactory`, `createUrl` and `patternUrl` props (names can be modified).